DigitalPassaport/DigitalPassaport/DataManagment/CoreDataManagment/CoreDataManager.swift

```diff
@@ -15,72 +15,59 @@ class CoreDataManager {
         self.context = context
     }
     
-    func saveAccountResponse(_ accountResponse: AccountResponse) {
+    func saveUserFromResponse(_ accountResponse: AccountResponse) {
+        context.perform {
+            let userEntity = accountResponse.user.toEntity(in: self.context)
+        }
+        do {
+            try self.context.save()
+        } catch {
+            print("Error saving data to Core Data: \(error)")
+        }
+    }
+    
+    func savePassFromResponse(_ accountResponse: AccountResponse) {
         context.perform {
-            let userItem = UserEntity(context: self.context)
-            userItem.firstName = accountResponse.user.firstName
-            userItem.lastName = accountResponse.user.lastName
-            userItem.image = accountResponse.user.image
-            userItem.email = accountResponse.user.email
-            
             for pass in accountResponse.passes {
-                let passItem = PassEntity(context: self.context)
-                passItem.id = pass.id
-                passItem.name = pass.name
-                passItem.descriptionCD = pass.description
-                passItem.icon = pass.icon
-            
-                userItem.addToPasses(passItem)
-            }
-            do {
-                try self.context.save()
-            } catch {
-                print("Error saving data to Core Data: \(error)")
+                let passEntity = pass.toEntity(in: self.context)
             }
         }
+        do {
+            try self.context.save()
+        } catch {
+            print("Error saving data to Core Data: \(error)")
+        }
     }
     
-    func fetchPasses() -> [PassEntity] {
+    func fetchPasses() -> [Pass] {
         let fetchRequest: NSFetchRequest<PassEntity> = PassEntity.fetchRequest()
-        
         do {
-             let passes = try self.context.fetch(fetchRequest)
-             var uniquePasses: [PassEntity] = []
-             var uniquePassNames: Set<String> = []
-             
-             for pass in passes {
-                 if !uniquePassNames.contains(pass.name ?? "") {
-                     uniquePassNames.insert(pass.name ?? "")
-                     uniquePasses.append(pass)
-                 }
-             }
-             return uniquePasses
+            let passEntities = try context.fetch(fetchRequest)
+            return passEntities.map { $0.toPass() }
```

Comment:
Return PassEntity but map it inside repository

Reply:
It's fixed